torch/onnx/_internal/fx/passes/readability.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+
+from typing import Dict, List, Sequence, Tuple, Union
+
+import torch
+from torch.onnx._internal import _beartype
+from torch.onnx._internal.fx import _pass, diagnostics
+
+
+class RestoreParameterAndBufferNames(_pass.Transform):
+    """Restore parameter and buffer names from original nn.module.
+
+    This pass is useful for readability of the exported ONNX graph. It restores the
+    parameter and buffer names from the original nn.module. For example, if the original
+    nn.module has a parameter named `root.linear.0.weight`, and the parameter is renamed to
+    `_param_constant9` by FX, this pass will rename it back.
+
+    This pass must be run after `Decompose` pass. Because this pass is expected to be called on
+    `fx.GraphModule` produced by `proxy_tensor.make_fx`, where all parameters and buffers
+    are registered at root level.
+    """
+
+    def __init__(
+        self,
+        diagnostic_context: diagnostics.DiagnosticContext,
+        fx_module: torch.fx.GraphModule,
+        nn_module: torch.nn.Module,
+    ):
+        super().__init__(diagnostic_context, fx_module)
+        self.original_module = nn_module
+
+    @_beartype.beartype
+    def _rename_param_and_buffer(
+        self,
+        diagnostic: diagnostics.Diagnostic,
+        nodes: Sequence[torch.fx.Node],
+        new_name: str,
+    ) -> None:
+        """Rename the parameter/buffer and replace corresponding nodes with new nodes of updated target."""
+        assert len(nodes) > 0, "`nodes` cannot be empty"
+        assert (
+            len({node.target for node in nodes}) == 1
+        ), "`nodes` must all have same `target`"
+        old_name = nodes[0].target
+        assert isinstance(old_name, str), f"Expected str, got type({old_name})"
+        # Parameter/buffer name cannot contain "."
+        # TODO(bowbao): fix #104670 and replace "." with "/" to avoid collision.
+        normalized_name = new_name.replace(".", "_")
+        attr_value = getattr(self.module, old_name)
+        setattr(self.module, normalized_name, attr_value)
+        delattr(self.module, old_name)
+        for node in nodes:
+            with self.module.graph.inserting_before(node):
+                new_node = self.module.graph.get_attr(normalized_name)
+                new_node.meta = node.meta
+                node.replace_all_uses_with(new_node)
+                self.module.graph.erase_node(node)
+        diagnostic.with_additional_message(
+            f"Renamed 'self.{old_name}' to 'self.{normalized_name}', "
+            f"normalized from original parameter name '{new_name}'."
+        )
+
+    def _run(self, *args, **kwargs) -> torch.fx.GraphModule:
+        """Restore parameter and buffer names from original module.
+
+        For each `get_attr` node, if the target is a str representing a parameter or buffer
+        under `self.module`, we rename the parameter or buffer to its original name.
```

Comment:
```suggestion
        under `self.fx_module`, we rename the parameter or buffer to its original name.
```
there is a bunch of old `self.module` going on here

Reply:
can't do anything, this is in base class